pkg/internal/server/grpc/get_metadata_allocated.go

```diff
@@ -114,22 +116,76 @@ func (s *Server) convertToCSIGetMetadataAllocatedRequest(ctx context.Context, re
 }
 
 func (s *Server) streamGetMetadataAllocatedResponse(ctx context.Context, clientStream api.SnapshotMetadata_GetMetadataAllocatedServer, csiStream csi.SnapshotMetadata_GetMetadataAllocatedClient) error {
+	var (
+		blockMetadataType   api.BlockMetadataType
+		lastByteOffset      int64
+		lastSize            int64
+		logger              = klog.FromContext(ctx)
+		numBlockMetadata    int
+		responseNum         int
+		volumeCapacityBytes int64
+	)
+
 	for {
 		csiResp, err := csiStream.Recv()
 		if err == io.EOF {
-			klog.FromContext(ctx).V(HandlerTraceLogLevel).Info("stream EOF")
+			logger.V(HandlerTraceLogLevel).WithValues(
+				"blockMetadataType", blockMetadataType.String(),
+				"lastByteOffset", lastByteOffset,
+				"lastSize", lastSize,
+				"lastResponseNum", responseNum,
+				"volumeCapacityBytes", volumeCapacityBytes,
+			).Info("stream EOF")
 			return nil
 		}
 
-		//TODO: stream logging with progress
-
 		if err != nil {
+			logger.WithValues(
+				"blockMetadataType", blockMetadataType.String(),
+				"lastByteOffset", lastByteOffset,
+				"lastSize", lastSize,
+				"lastResponseNum", responseNum,
+				"volumeCapacityBytes", volumeCapacityBytes,
+			).Error(err, msgInternalFailedCSIDriverResponse)
 			return s.statusPassOrWrapError(err, codes.Internal, msgInternalFailedCSIDriverResponseFmt, err)
 		}
 
+		responseNum++
+
 		clientResp := s.convertToGetMetadataAllocatedResponse(csiResp)
+		blockMetadataType = clientResp.BlockMetadataType
+		volumeCapacityBytes = clientResp.VolumeCapacityBytes
+		numBlockMetadata = len(clientResp.BlockMetadata) - 1
+		if numBlockMetadata >= 0 {
+			lastByteOffset = clientResp.BlockMetadata[numBlockMetadata].ByteOffset
+			lastSize = clientResp.BlockMetadata[numBlockMetadata].SizeBytes
+		}
+
+		if logger.V(HandlerTraceLogLevel).Enabled() {
```

Comment:
I am concerned whether we need another higher tracing level to display this data.